core/cache/model.go

```diff
@@ -65,6 +65,44 @@ func (m *Model) WatchConfig(keys ...string) *ConfigWatcher {
 	return newConfigWatcher(keys, m.hashCache, m.hub, m.topic(modelConfigChange))
 }
 
+func (m *Model) mark() int {
+	var result int
+	m.mu.Lock()
+	for _, app := range m.applications {
+		app.mark()
+		result++
+	}
+	m.mu.Unlock()
```

Comment:
What about machines and charms?
And the model itself?
And units... although I had kinda thought that the application would hold the units for itself.

Reply:
Oversight.